src/Microsoft.Health.Fhir.Core/Features/Operations/BulkUpdate/BulkUpdateOrchestratorJob.cs

```diff
@@ -0,0 +1,295 @@
+
+                    var enqueued = groupJobs.Where(x => x.Id != jobInfo.Id) // exclude coord
+                                            .Select(x => JsonConvert.DeserializeObject<BulkUpdateDefinition>(x.Definition))
+                                            .Where(x => x.EndSurrogateId != null) // This is to handle current mock tests. It is not needed but does not hurt.
+                                            .GroupBy(x => x.Type)
+                                            .ToDictionary(x => x.Key, x => x.Max(r => long.Parse(r.EndSurrogateId)));
+
+                    await Parallel.ForEachAsync(resourceTypes, new ParallelOptions { MaxDegreeOfParallelism = CoordinatorMaxDegreeOfParallelization, CancellationToken = cancellationToken }, async (type, cancel) =>
+                    {
+                        var startId = globalStartId;
+                        if (enqueued.TryGetValue(type, out var max))
+                        {
+                            startId = max + 1;
+                        }
+
+                        var rows = 1;
+                        while (rows > 0)
+                        {
+                            var definitions = new List<BulkUpdateDefinition>();
+                            var ranges = await searchService.Value.GetSurrogateIdRanges(type, startId, globalEndId, surrogateIdRangeSize, NumberOfParallelRecordRanges, true, cancel);
+                            foreach (var range in ranges)
+                            {
+                                if (range.EndId > startId)
+                                {
+                                    startId = range.EndId;
+                                }
+
+                                _logger.LogJobInformation(jobInfo, "Creating bulk update definition (1).");
+                                var processingDefinition = CreateProcessingDefinition(definition, searchService.Value, cancellationToken, type, continuationToken: null, startSurrogateId: range.StartId.ToString(), endSurrogateId: range.EndId.ToString(), globalStartSurrogateId: globalStartId.ToString(), globalEndSurrogateId: globalEndId.ToString());
+                                definitions.Add(processingDefinition);
+                            }
+
+                            startId++; // make sure we do not intersect ranges
+
+                            rows = definitions.Count;
+                            if (rows > 0)
+                            {
+                                _logger.LogJobInformation(jobInfo, "Enqueuing bulk update job (1).");
+                                await _queueClient.EnqueueAsync(QueueType.BulkUpdate, cancel, groupId: jobInfo.GroupId, definitions: definitions.ToArray());
+                            }
+                        }
+                    });
+                }
+                else if (definition.IsParallel)
```

Comment:
This case isn't rerun safe. If the orchestrator job crashes in the middle of executing it may create multiple identical processing jobs.